### PR TITLE
TEET-1585 Cost Items location reference - Fetch relevant roads

### PR DESCRIPTION
**Pull request by outamaa**
This PR replaces the text field selection of road number for a cost item with a dropdown of relevant roads.

### CODE REVIEW

File: app/backend/dev-src/user.clj [hunk 1]
```diff
@@ -336,8 +337,15 @@
...
-(def local-command tu/local-command)
+(defn local-query [& args]
+  (binding [tu/*connection* (db-connection)
+            tu/*asset-connection* (asset-connection)]
+    (apply tu/local-query args)))
+
+(defn local-command [& args]
+  (binding [tu/*connection* (db-connection)
+            tu/*asset-connection* (asset-connection)]
+    (apply tu/local-command args)))
```
Comment (outamaa) [inline app/backend/dev-src/user.clj:348]: Changed these to use the local dev db connections as they used to do.

File: app/backend/src/clj/teet/asset/asset_queries.clj [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
+            [teet.db-api.core :as db-api :refer [defquery]]
+            [teet.environment :as environment]
             [teet.localization :as localization]
             [teet.log :as log]
-            [teet.user.user-db :as user-db]))
+            [teet.project.project-db :as project-db]
+            [teet.road.road-query :as road-query]
+            [teet.transit :as transit]
+            [teet.user.user-db :as user-db]
+            [teet.util.euro :as euro]))
```
Comment (outamaa) [inline app/backend/src/clj/teet/asset/asset_queries.clj:16]: Sorted alphabetically.

File: app/backend/src/clj/teet/map/map_services.clj [hunk 3]
```diff
@@ -112,12 +112,26 @@
...
                             request-delay (or custom-parse-feature parse-feature))))))
 
-(defn wfs-get-feature [{:keys [wfs-url]} typename ogc-filter]
+(defn- propertyName->string
+  "Transforms a vec of keyword property names into a comma separated list:
+   [:a :b :c] -> \"a,b,c\""
+  [propertyName]
+  (->> propertyName
+       (map name)
+       (str/join ",")))
```
Comment (outamaa) [inline app/backend/src/clj/teet/map/map_services.clj:121]: This now allows one to restrict the properties fethed from WFS. For example, the road geometries are irrelevant for the task at hand.

File: app/backend/src/clj/teet/map/map_services.clj [hunk 4]
```diff
@@ -112,12 +112,26 @@
...
+   typename                                     ;; The `typeName` of the feature
+   ogc-filter                                   ;; the OGC filter in hiccup format
+   & [{:keys [propertyName] :as _options-map}]] ;; a map of other options; `propertyName` is a vector of property keywords to return
   (let [payload [:wfs:GetFeature {:xmlns:wfs "http://www.opengis.net/wfs"
                                   :xmlns:gml "http://www.opengis.net/gml"
                                   :version "1.1.0"
                                   :service "WFS"}
-                 [:wfs:Query {:typeName typename}
+                 [:wfs:Query (merge {:typeName typename}
+                                    (when propertyName {:propertyName (propertyName->string propertyName)}))
```
Comment (outamaa) [inline app/backend/src/clj/teet/map/map_services.clj:134]: Documentation and changes related to `propertyName`

File: app/backend/src/clj/teet/project/project_commands.clj [hunk 5]
```diff
@@ -200,14 +197,6 @@
...
          :error :permission-already-granted}))))
 
-(defn- thk-id->integration-id
-  "given project's `:thk.project/id`, fetch the said project's
-  `:integration/id` number form"
-  [db thk-id]
-  (-> (du/entity db [:thk.project/id thk-id])
-      :integration/id
-      integration-id/uuid->number))
-
```
Comment (outamaa) [inline app/backend/src/clj/teet/project/project_commands.clj:210]: Moved to `project-db`.

File: app/backend/src/clj/teet/road/road_query.clj [hunk 6]
```diff
@@ -232,11 +232,20 @@
...
+  project road object (radius 50m)"
+  [ctx integration-id buffer-meters]
+  {:pre [integration-id]}
+  (let [road-object-search-geometry (entity-features/entity-search-area-gml ctx
+                                                                            integration-id
+                                                                            buffer-meters)]
+    (->> (map-services/fetch-intersecting-objects-of-type ctx
+                                                          "ms:teeosa"
+                                                          road-object-search-geometry
+                                                          {:propertyName [:ms:tee_number :ms:soidutee_nr :ms:nimi]})
```
Comment (outamaa) [inline app/backend/src/clj/teet/road/road_query.clj:246]: Change to `return-properties`? It's not clear whether this refers to -> or <-

File: app/backend/src/clj/teet/map/map_services.clj [hunk 7]
```diff
@@ -112,12 +112,26 @@
...
+  (->> return-properties
+       (map name)
+       (str/join ",")))
+
+(defn wfs-get-feature
+  "Returns features of type `typename`, filtered by `ogc-filter`."
+  [{:keys [wfs-url]}
+   typename                                     ;; The `typeName` of the feature
+   ogc-filter                                   ;; the OGC filter in hiccup format
+   & [{:keys [return-properties] :as _options-map}]] ;; a map of other options; `return-properties` is a vector of property keywords to return
```
Comment (tatut) [inline app/backend/src/clj/teet/map/map_services.clj:128]: should this comment instead be in the docstring as it describes input parameters

**Pull request merged**